File: backend-spring/domain/src/main/java/apptemplate/domain/entities/UploadedFile.java
package apptemplate.domain.entities;

import lombok.AllArgsConstructor;
import lombok.experimental.SuperBuilder;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

/**
 * UploadedFile entity with business logic.
 */
@Getter
@Setter
@SuperBuilder
@NoArgsConstructor
@AllArgsConstructor
public class UploadedFile extends AuditableEntity {

    private String fileName;
    private String originalFileName;
    private String contentType;
    private Long fileSize;
    private String storagePath;
    private String description;
    private String category;
    private boolean isPublic;



    /**
     * Updates file metadata.
     */
    public void update(String description, String category, Boolean isPublic) {
        if (description != null) this.description = description;
        if (category != null) this.category = category;
        if (isPublic != null) this.isPublic = isPublic;
    }
}

File: backend/dotnet/clean-architecture/minimal/src/Core/App.Template.Domain/Entities/UploadedFile.cs
namespace AppTemplate.Domain.Entities;

public class UploadedFile : AuditableEntity
{
    public string FileName { get; private set; } = null!;
    public string OriginalFileName { get; private set; } = null!;
    public string ContentType { get; private set; } = null!;
    public long FileSize { get; private set; }
    public string StoragePath { get; private set; } = null!;
    public string? Description { get; private set; }
    public string? Category { get; private set; }
    public bool IsPublic { get; private set; }

    private UploadedFile() { }

    public UploadedFile(
        string fileName,
        string originalFileName,
        string contentType,
        long fileSize,
        string storagePath,
        string? description = null,
        string? category = null,
        bool isPublic = false)
    {
        FileName = fileName;
        OriginalFileName = originalFileName;
        ContentType = contentType;
        FileSize = fileSize;
        StoragePath = storagePath;
        Description = description;
        Category = category;
        IsPublic = isPublic;
    }

    public void Update(string? description, string? category, bool? isPublic)
    {
        if (description != null) Description = description;
        if (category != null) Category = category;
        if (isPublic.HasValue) IsPublic = isPublic.Value;
    }
}
